src/sdk/sdkSession.ts

```diff
@@ -564,46 +564,66 @@ class SDKSession {
     const processedTxs: any[] = [];
     const txs = JSON.parse(JSON.stringify(this.btcTxs));
     txs.forEach((tx) => {
-      tx.recipient = tx.outputs[0].addr;
-      tx.incoming = allAddrs.indexOf(tx.recipient) > -1;
+      // Determine if this is an outgoing transaction or not based on inputs.
+      // If *any* input comes from one of our addresses, we consider the 
+      // transaction to be "outgoin".
+      let outgoing = false;
+      tx.inputs.forEach((input) => {
+        if (allAddrs.indexOf(input.addr) > -1) {
+          outgoing = true;
+        }
+      })
+      tx.incoming = !outgoing;
```

Comment:
I don't think it's worth holding up the PR and retesting everything for this, but this type of logic can also be written like this: 

```ts
tx.incoming = tx.inputs.every(input => allAddrs.indexOf(input.addr) > -1)
```

which could be a bit easier to parse for some people.

Reply:
[2211973](https://github.com/GridPlus/lattice-manager/pull/219/commits/221197375f71b9b0441d0868439625d6cc3ddddb)